spyder/widgets/sourcecode/base.py

```diff
@@ -317,43 +319,61 @@ def set_palette(self, background, foreground):
 
 
     #------Extra selections
-    def extra_selection_length(self, key):
-        selection = self.get_extra_selections(key)
-        if selection:
-            cursor = self.extra_selections_dict[key][0].cursor
-            selection_length = cursor.selectionEnd() - cursor.selectionStart()
-            return selection_length
-        else:
-            return 0
-
     def get_extra_selections(self, key):
+        """Return editor extra selections.
+
+        Args:
+            key (str) name of the extra selections group
+
+        Returns:
+            list of sourcecode.api.TextDecoration.
+        """
         return self.extra_selections_dict.get(key, [])
 
     def set_extra_selections(self, key, extra_selections):
+        """Set extra selections for a key.
+
+        Also assign draw orders to leave current_cell and current_line
+        in the backgrund (and avoid them to cover other decorations)
+
+        NOTE: This will remove previous decorations added to  the same key.
+
+        Args:
+            key (str) name of the extra selections group.
+            extra_selections (list of sourcecode.api.TextDecoration).
+        """
+        # current line has to be highlighted first
+        draw_orders = {'current_cell': 0,
+                       # draw order = 1 is used in codefolding

```

Comment:
Why `0` can't be used for code folding?

Reply:
I put 1 to make it appear in front of current cell, but behind other selections.

I'm not sure, what is the best order, codefolding in the back? Or maybe in the front and be a little transparent?

Reply:
This needs to be explained when you define the constants I mentioned above.